### ISSUE TITLE
Can image augmentation without cutting the augmented result?

### ISSUE THREAD

**Issue by hellozgm** [aleju/imgaug#141]
Many image augmentation method such as Affine rotation and PerspectiveTransform may generate the  cutting augmented result. Here is an example, after I rotate the lenna image by 45 degree, the default Affine method generate the cutting edge result:
![lenna_rotation](https://user-images.githubusercontent.com/9148863/41636095-0f4a7d64-747f-11e8-8aa8-d7f8ca257fdc.jpg)

But I want the rotation keep all augmented image edges, like this:

![lenna_rotation_scale](https://user-images.githubusercontent.com/9148863/41636339-7192357e-7480-11e8-8722-397973041ce5.jpg)

So, are there any way to do so? Many thanks!

**Comment by aleju** [comment]
No, that's currently not possible. I think the waiting pull request would have added something in that direction to scikit-image's affine warping function, which then could have been added to Affine, but so far it was not merged to scikit-image. So the only way here to do this is to manually compute scalings that don't remove image content given sampled rotation and shearing values. But would require a partial rewrite of Affine, as scale, rotation and shear are sampled independently from each other.

For PerspectiveTransform it would be even harder as that augmenter currently samples a subpolygon within the image and "zooms into" that, thereby removing anything outside of the polygon.

**Comment by hellozgm** [comment]
OK, I would try your suggestion, thanks for your reply.
